src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -268,6 +269,9 @@ protected void doRun() {
                                     .withType(TransportRequestOptions.Type.RECOVERY)
                                     .withTimeout(recoverySettings.internalActionTimeout());
 
+                            // When we last consulted RateLimiter:
+                            long lastPauseCheckBytes = 0;
```

Comment:
this needs to be out of the file loop, no? which also means we need a total read count